pkg/plugins/policies/core/xds/meshroute/clusters.go

```diff
@@ -78,9 +79,26 @@ func GenerateClusters(
 				} else {
 					if msName := service.MeshServiceName(); len(msName) > 0 {
 						identities := meshCtx.MeshServiceIdentity[msName].Identities
+						ms := meshCtx.MeshServiceIdentity[msName].Resource
+						name := ms.Meta.GetName()
+						// we need to use original name and namespace for services that were synced from another cluster
+						if dn := ms.GetMeta().GetLabels()[mesh_proto.DisplayName]; dn != "" {
```

Comment:
Feels like "original core name" should be a label too ideally

Reply:
Maybe, but I'm not sure about introducing yet another name label :/